src/enforcer.rs

```diff
@@ -154,20 +169,20 @@ impl Enforcer {
     /// #[cfg(feature = "runtime-async-std")]
     /// #[async_std::main]
     /// async fn main() {
-    ///     let e = Enforcer::new("examples/basic_model.conf", "examples/basic_policy.csv").await.unwrap();
-    ///     assert_eq!(true, e.enforce(&["alice", "data1", "read"]).unwrap());
+    ///     let mut e = Enforcer::new("examples/basic_model.conf", "examples/basic_policy.csv").await.unwrap();
+    ///     assert_eq!(true, e.enforce(&["alice", "data1", "read"]).await.unwrap());
     /// }
     ///
     /// #[cfg(feature = "runtime-tokio")]
     /// #[tokio::main]
     /// async fn main() {
-    ///     let e = Enforcer::new("examples/basic_model.conf", "examples/basic_policy.csv").await.unwrap();
-    ///     assert_eq!(true, e.enforce(&["alice", "data1", "read"]).unwrap());
+    ///     let mut e = Enforcer::new("examples/basic_model.conf", "examples/basic_policy.csv").await.unwrap();
+    ///     assert_eq!(true, e.enforce(&["alice", "data1", "read"]).await.unwrap());
     /// }
     /// #[cfg(all(not(feature = "runtime-async-std"), not(feature = "runtime-tokio")))]
     /// fn main() {}
     /// ```
-    pub fn enforce<S: AsRef<str>>(&self, rvals: &[S]) -> Result<bool> {
+    async fn enforce<S: AsRef<str> + Send + Sync>(&mut self, rvals: &[S]) -> Result<bool> {
```

Comment:
I found no mutation in `enforce` method, do we really need to use `&mut self` here?